sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkBatchOperation.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Analytics.Synapse.Spark.Models;
+
+namespace Azure.Analytics.Synapse.Spark
+{
+    /// <summary>
+    /// A long-running operation for:
+    /// <see cref="SparkBatchClient.StartCreateSparkBatchJobAsync(SparkBatchJobOptions, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartCreateSparkBatchJob(SparkBatchJobOptions, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartGetSparkBatchJobAsync(int, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartGetSparkBatchJob(int, bool?, CancellationToken)"/>
+    /// </summary>
+    public class SparkBatchOperation : Operation<SparkBatchJob>
+    {
+        private static readonly TimeSpan s_defaultPollingInterval = TimeSpan.FromSeconds(5);
+
+        private readonly ClientDiagnostics _diagnostics;
+
+        private readonly SparkBatchClient _client;
+        private readonly SparkBatchJob _value;
+        private Response<SparkBatchJob> _response;
+        private bool _completed;
+
+        private static List<string> SessionSubmissionFinalStates = new List<string>
```

Comment:
are these values that are set in the swagger? or is there a way they can be set in the swagger so instead of having a local type, there can be a generated one?

Reply:
Does not appear to be:

```
$ git grep "dead"
preview/2019-11-01-preview/sparkJob.json:        "deadAt": {
preview/2019-11-01-preview/sparkJob.json:          "description": "time that at which \"dead\" livy state was first seen.",
preview/2019-11-01-preview/sparkJob.json:        "deadAt": {
donblas:~/Programming/azure-rest-api-specs/specification/synapse/data-plane/Microsoft.Synapse 
$ git grep "killed"
preview/2019-11-01-preview/sparkJob.json:        "killedAt": {
preview/2019-11-01-preview/sparkJob.json:          "description": "the time that at which \"killed\" livy state was first seen.",
preview/2019-11-01-preview/sparkJob.json:        "killedAt": {
```

Reply:
I will expect to see this as enums in the swagger. Any idea why not? I mean, not blocking your PR but it will be interesting to push on the service to understand that a bit more

Reply:
my worry is that this is not mapped with the service at all. So if the service changes/adds/etc our code will need to be manually updated